docs/defi/flow-yield-vaults/architecture.md

```diff
@@ -0,0 +1,375 @@
+---
+title: Architecture Overview
+sidebar_position: 2
+---
+
+# Architecture Overview
+
+Flow Yield Vaults (FYV) is built on a modular architecture that separates concerns between user position management, yield strategy implementation, and automated rebalancing. This document explains the core components and how they interact to create an automated leveraged yield farming system.
+
+## System Architecture
+
+```mermaid
+graph TB
+    User[User Account] -->|owns| YVM[YieldVaultManager]
+    YVM -->|contains| YV1[YieldVault 1]
+    YVM -->|contains| YV2[YieldVault 2]
+
+    YV1 -->|delegates to| Strategy[Strategy Implementation]
+    YV1 -->|manages| AB[AutoBalancer]
+    YV1 -->|holds capability| Pos[ALP Position]
```

Comment:
We mention capabilities a few times in the article but its not clear what those are for people not in our ecosystem, perhaps hyperlinking to the capabilitiy docs or explaining it at the first mention